.travis.yml

```diff
@@ -5,20 +5,34 @@ language: php
 php:
     - 7.2
 
-branches:
-    only:
-        - master
-
-cache:
-    directories:
-        - $HOME/.composer/cache/files
-
 before_install:
-    - phpenv config-rm xdebug.ini
+    - phpenv config-rm xdebug.ini || true
 
 install:
     - travis_retry composer install --classmap-authoritative --no-suggest --prefer-dist
 
 script:
     - vendor/bin/phpunit
-    - vendor/bin/phpcs
+
+jobs:
+    include:
+
+        - stage: Code Quality
+          env: CODING_STANDARDS
+          script:
+              - vendor/bin/phpcs -p
+
+stages:
+    - Test
+    - name: Code Quality
+      if: type = pull_request
+
+cache:
+    directories:
+        - $HOME/.composer/cache/files
+
+if: |
+    branch = master OR \
+    branch =~ /^(?:[0-9]|[1-9][0-9]*)\.(?:[0-9]|[1-9][0-9]*)$/ OR \
+    tag IS present OR \
```

Comment:
Not sure if we're having version branches/tags given the schemas aren't versioned? Though is the repo itself version?

Reply:
An explicit version is needed to propagate additions even if there is no break of compatibility e.g. need to go on `0.2` rather than `0.1` if I want to use a new schema

Reply:
Depends if this is meant to be includable as a library, or accessible only over HTTP.